airbyte-webapp/src/components/ui/PillSelect/PillSelect.tsx

```diff
@@ -22,11 +21,11 @@ export const PillSelect: React.FC<PillSelectProps> = ({ className, ...props }) =
             ? value.map(({ label }: { label: string }) => label).join(", ")
             : value.label
           : "";
-
         return (
           <Tooltip
             control={
               <PillButton
+                labels={Array.isArray(value?.label) ? value?.label : [value?.label]}
```

Comment:
It would be better if the PillButton can distinguish between the array and non-arrays